Hints + PROGREssbars (son del PSOE) + Favicon + Buttons visibility + filters

ignapas opened the pull request:
<!--  **WIP-** prefix in title if still work in progress -->

## What do these changes do?

<!-- Please give a short brief about these changes. -->
* Progress bars are fixed.
* Add a favicon (little icon in the browser tab), now done properly (no JS)
* Managed visibility of Group/Ungroup buttons
* Fixed service browser's filters

## Related issue number

<!-- Please add #issues -->
#1281 #1282 

## How to test

<!-- Please explain how this can be tested. Also state wether this PR needs a full rebuild, database changes, etc. -->
### @odeimaiz I changed some code to reuse the same ``StudyEditor`` between different studies. Going back to the Dashboard and opening another study now uses the same ``StudyEditor`` instance. I did it because we were attaching a lot of event handlers to a lot of widgets and many copies of the callbacks where being executed. But this could break some stuff... please review :)
* Play with selection (in workbench and node tree) and check buttons' visibility

## Screenshot
![cast4pedro](https://user-images.githubusercontent.com/4764217/75043669-bbd56580-54c0-11ea-8295-506a535bb574.gif)


## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [x] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

odeimaiz commented on services/web/client/source/class/osparc/desktop/ControlsBar.js:
```diff
@@ -141,13 +150,32 @@ qx.Class.define("osparc.desktop.ControlsBar", {
+      const selectedNodes = msg.getData();
+      let groupBtnVisibility = "excluded";
+      let ungroupBtnVisibility = "excluded";
+      if (selectedNodes.length > 1) {
```
if (selectedNodes.length)

Grouping a single node should be possible

ignapas commented:
New changes to test:
* Fixed problems with hints
* Applied permissions to editing node and study names from the ``EditLabel``
* Added permissions to (user clearance)
  * Tags
  * Grouping
* Really solved the progressbar issue

odeimaiz commented on services/web/client/source/class/osparc/data/Permissions.js:
```diff
@@ -138,14 +138,17 @@ qx.Class.define("osparc.data.Permissions", {
           "preferences.user.update",
           "preferences.token.create",
           "preferences.token.delete",
+          "preferences.tag",
```
The convention we are following for the permissions is:
scope.resource.action

ignapas replied:
does it make sense to leave it to block all actions inside the preferences view? i am not showing the tab at all...

pcrespov replied:
@ignapas Please sync from time to time the front-end permissions with [those defined in the backend](https://github.com/ITISFoundation/osparc-simcore/blob/857de39ba6438e97eda2d913e15aedd470598772/services/web/server/src/simcore_service_webserver/security_roles.py)
